### PR TITLE
switch from json.org to jackson - minimal version

### PR DESCRIPTION
**Pull request by michaz**
Replaces json.org with jackson. Basically 1:1, without making much use of its features yet.

This does _not_ mean we can't switch to Gson. On the contrary, those two are very similar, and at the moment it would be search+replace, I think.

### CODE REVIEW

**Comment by karussell** [comment]
Thanks! And +1 for jackson for now ...

File: web/src/main/java/com/graphhopper/http/GHBaseServlet.java [hunk 1]
```diff
@@ -65,33 +71,33 @@ protected void writeJson(HttpServletRequest req, HttpServletResponse res, JSONOb
...
+                writeResponse(res, callbackName + "(" + objectMapper.writeValueAsString(json) /*.toString(2)*/ + ")");
             else
-                writeResponse(res, callbackName + "(" + json.toString(0) + ")");
+                writeResponse(res, callbackName + "(" + objectMapper.writeValueAsString(json) /*.toString(0)*/ + ")");
 
         } else {
             res.setContentType("application/json");
             if (debug)
-                writeResponse(res, json.toString(2));
+                writeResponse(res, objectMapper.writeValueAsString(json) /*.toString(2)*/);
```
Comment (karussell) [inline web/src/main/java/com/graphhopper/http/GHBaseServlet.java]: Here we would need an additional "pretty" objectMapper: 

ObjectMapper prettyOM = new ObjectMapper();
prettyOM.enable(SerializationFeature.INDENT_OUTPUT);

Comment (michaz) [inline web/src/main/java/com/graphhopper/http/GHBaseServlet.java]: Ah, this can also be done "by request".

Comment (michaz) [inline web/src/main/java/com/graphhopper/http/GHBaseServlet.java]: The super-advanced DI-way would probably be to bind() a request-scoped ``ObjectWriter`` which is debug or non-debug, and then inject that configured ``ObjectWriter`` everywhere..

Comment (michaz) [inline web/src/main/java/com/graphhopper/http/GHBaseServlet.java]: Apparently, this is _the_ use case. From the doc:

> Construction of new ObjectReaders and ObjectWriters is a very light-weight operation so it is usually appropriate to create these on per-call basis, as needed, for configuring things like optional indentation of JSON.

**Comment by karussell** [comment]
Thanks, I will merge. (this time I can squash I guess?)

**Comment by karussell** [comment]
This then fixes #773 I think (?) and would you mind to do similar work for the map matching :) ?

**Comment by michaz** [comment]
> Thanks, I will merge. (this time I can squash I guess?)
Yes please.

**Comment by karussell** [comment]
just fyi: I just stumbled over [a possibility](https://stackoverflow.com/a/33578478/194609) which avoids a jackson dependency in the core module.

**Pull request merged**